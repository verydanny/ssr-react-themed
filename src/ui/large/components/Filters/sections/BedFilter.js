import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { themed } from 'react-themed-too'
import autobind from 'autobind-decorator'
import cn from 'classnames'
import omit from 'lodash/omit'
import bedOptions from 'config/refinements/bedFilterLarge'
import { Dropdown } from '@rentpath/react-ui-core'
import MenuWrapper from './MenuWrapper'
import FiltersButton from './FiltersButton'

@themed('*')
export default class BedFilter extends Component {
  static propTypes = {
    buttonText: PropTypes.string,
    theme: PropTypes.object,
    onChange: PropTypes.func.isRequired,
  }

  static defaultProps = {
    buttonText: 'Beds',
    theme: {},
  }

  @autobind
  handleOptionChange(options) {
    bedOptions.map(option => {
      if (option.label === options.props.children) {
        this.props.onChange(option.value)
      }
      return option.value
    })
  }

  @autobind
  handleOptionSelection() {
    const { theme, buttonText } = this.props
    const listItems = bedOptions.map(option => (
      <span
        className={
          option.label === buttonText ?
            theme['Filters_DropdownOption-selected'] : ''
        }
        data-tag_selection={option.selection}
      >
        {option.label}
      </span>
    ))
    return listItems
  }

  render() {
    const {
      theme,
      buttonText,
    } = this.props

    return (
      <div
        data-tag_item="filters_bedrooms"
        data-tag_section="quick_filters"
        className={theme.Filters_DropdownWrapper}
      >
        <Dropdown
          anchorField={props => (
            <FiltersButton
              buttonText={buttonText}
              {...omit(props, 'className')}
            />
          )}
          className={cn(theme['Filters_DropdownContainer-beds'], theme.Filters_Text)}
        >
          <MenuWrapper
            options={this.handleOptionSelection()}
            className={cn(theme.Filters_Dropdown, theme['Filters_Dropdown-beds'])}
            setButtonText={this.handleOptionChange}
          />
        </Dropdown>
      </div>
    )
  }
}
