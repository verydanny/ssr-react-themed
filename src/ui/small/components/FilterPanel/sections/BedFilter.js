import React from 'react'
import PropTypes from 'prop-types'
import { Counter } from '@rentpath/react-ui-core'
import CounterIcon from './CounterIcon'

const text = (value, min, max) => num => {
  if (value === max) {
    return `${num} +`
  } else if (value === 0) {
    return 'Studio'
  } else if (value > 1) {
    return `${num}`
  } else if (!value || value === min) {
    return 'Any'
  }
  return `${num}`
}

const BedFilter = ({ onClick, label, max, min, value }) => (
  <Counter
    count={value}
    decrementOperator={
      <CounterIcon
        disabled={value === min || (value < 0)}
        name="subtract"
      />
    }
    incrementOperator={
      <CounterIcon
        disabled={value === max}
        name="add"
      />
    }
    label={label}
    text={text(value, min, max)}
    max={max}
    min={min}
    onClick={onClick}
    data-tid="beds-counter"
  />
)

BedFilter.propTypes = {
  label: PropTypes.string,
  onClick: PropTypes.func,
  max: PropTypes.number,
  min: PropTypes.number,
  value: PropTypes.number,
}

BedFilter.defaultProps = {
  value: -1,
  min: -1,
  max: 4,
  label: 'Bedrooms',
}

export default BedFilter
