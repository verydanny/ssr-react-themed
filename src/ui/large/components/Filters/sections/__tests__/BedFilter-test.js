import React from 'react'
import { mount } from 'enzyme'
import BedFilter from '../BedFilter'

const theme = {
  Filters_Text: 'Filters_DropdownContainer-beds',
  'Filters_DropdownOption-selected': 'Filters_DropdownOption-selected',
}

function setup() {
  const props = {
    theme,
    buttonText: 'Studio',
    onChange: jest.fn(),
  }

  const wrapper = mount(<BedFilter {...props} />)
  const buttonText = wrapper.find(`.${theme.Filters_Text}`).first().text()
  const button = wrapper.find('Button')

  return {
    wrapper,
    buttonText,
    button,
  }
}

describe('Bed Filter', () => {
  const test = setup()
  const props = test.wrapper.props()

  it('contains text equal to buttonText props', () => {
    expect(test.buttonText).toBe(props.buttonText)
  })

  it('calls onChange when option is click', () => {
    test.button.simulate('click')
    const option = test.wrapper.find('ListItem').first()
    option.simulate('click')
    expect(props.onChange).toBeCalled()
  })

  it('contains expected options in dropdown', () => {
    test.button.simulate('click')
    const option = test.wrapper.find('ListItem')
    expect(option.at(0).text()).toBe('Any')
    expect(option.at(5).text()).toBe('4+ Beds')
  })
})

describe('handleOptionSelection method', () => {
  const test = setup()

  it('adds the selected classname to correct option', () => {
    const result = test.wrapper.find('BedFilter').instance().handleOptionSelection()
    expect(result).toMatchSnapshot()
  })
})
