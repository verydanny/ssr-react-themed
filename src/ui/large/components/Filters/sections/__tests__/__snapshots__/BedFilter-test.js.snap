// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handleOptionSelection method adds the selected classname to correct option 1`] = `
Array [
  <span
    className=""
    data-tag_selection="any"
  >
    Any
  </span>,
  <span
    className="Filters_DropdownOption-selected"
    data-tag_selection="0beds"
  >
    Studio
  </span>,
  <span
    className=""
    data-tag_selection="1bed"
  >
    1 Bed
  </span>,
  <span
    className=""
    data-tag_selection="2beds"
  >
    2 Beds
  </span>,
  <span
    className=""
    data-tag_selection="3beds"
  >
    3 Beds
  </span>,
  <span
    className=""
    data-tag_selection="4beds"
  >
    4+ Beds
  </span>,
]
`;
